modules/kubernetes-addons/aws-load-balancer-controller/locals.tf

```diff
@@ -40,7 +40,7 @@ locals {
   }
 
   irsa_config = {
-    kubernetes_namespace              = "kube-system"
+    kubernetes_namespace              = local.helm_config["namespace"]
     kubernetes_service_account        = local.service_account_name
     create_kubernetes_namespace       = false
```

Comment:
This needs to be changed to `create_kubernetes_namespace = true`  when we use namespace other than `kube-system` to ensure namespace is created by the IRSA module

Reply:
@askulkarni2 May be we need to handle this in IRSA module to look for `kube-system`  when `create_kubernetes_namespace=true`.

See below 

```hcl
resource "kubernetes_namespace_v1" "irsa" {
  count = var.create_kubernetes_namespace && var.kubernetes_namespace != "kube-system" ? 1 : 0
  metadata {
    name = var.kubernetes_namespace

    labels = {
      "app.kubernetes.io/managed-by" = "terraform-eks-blueprints"
    }
  }
}
```

Reply:
> This needs to be changed to `create_kubernetes_namespace = true` when we use namespace other than `kube-system` to ensure namespace is created by the IRSA module

I'm not sure I follow the logic. In my use-case, the namespace already exists. So this wouldn't work for me. That isn't done in other instances such as for the [kubernetes_dashboard module](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/f4a1a00bd92f5fe2e7cd44d6883cd90d291d1f9b/modules/kubernetes-addons/kubernetes-dashboard/locals.tf#L38). 

I would suggest the `create_kubernetes_namespace` option should be configurable for each of the given addon modules. Perhaps as part of the [`addon_context`](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/f4a1a00bd92f5fe2e7cd44d6883cd90d291d1f9b/modules/kubernetes-addons/aws-load-balancer-controller/variables.tf#L13). 

I'd be happy to look at that, but this is not the intent of this PR. I'm trying to draw these modules level with others such as the `kubernetes_dashboard` that would work with an existing namespace that is _not_ `kube-system`. This behaviour did previously work.

Reply:
Thanks @bobdoah I appreciate your logic. Let me go through in detail

Namespace creation is handled only by EKS Blueprints module at this moment apart from existing system namespaces like `kube-system`. Users can pass any name as namespace through the below configuration. 

```hcl
  enable_aws_load_balancer_controller = true
  # Optional  
  aws_load_balancer_controller_helm_config = {
    name                       = "aws-load-balancer-controller"
    chart                      = "aws-load-balancer-controller"
    repository                 = "https://aws.github.io/eks-charts"
    version                    = "1.3.1"
    namespace                  = "my_alb_ns"
    values = [templatefile("${path.module}/values.yaml", {})]
  }
```
User defined `my_alb_ns` namespace overrides the locally defined namespace. When we specify `kubernetes_namespace = local.helm_config["namespace"]` and `create_kubernetes_namespace = false`  then the above example will introduce a defect as `my_alb_ns` wont be created due to `false` condition. 

Yes, I just noticed that  `kubernetes-dashboard` config is incorrect as well. I think the solution for this would be as follows

- We will discuss about your suggestion allowing users to define `create_kubernetes_namespace` through helm config

Reply:
@vara-bonthu I understand what you're saying, but that doesn't fit my use case. The namespace has already been created externally, it's not `kube-system`. I have no control over this, sadly. If I set it to `true` I still won't be able to use this module, as-is. 

I have pushed a further commit. I think this is the simplest way I can satisfy both yours and my use cases. The namespace will be created, if it is not the default `kube-system`. The `create_kubernetes_namespace` option can be overridden by supplying a custom `irsa_config` to the module. The contents will be merged, in the same way as is done for the `helm_config`.